src/BalanceTx.purs

```diff
@@ -109,22 +103,46 @@ import Undefined (undefined)
 -- These derivations may need tweaking when testing to make sure they are easy
 -- to read, especially with generic show vs newtype show derivations.
 data BalanceTxFailure
```

Comment:
I think the number of type/data constructors for different error conditions might be getting a bit out of hand. Some of these can just have `String`s giving more context instead of two additional levels of data constructors (e.g. most of the `*Failure` -> `*Reason` newtypes)

Reply:
Will look to simplify errors.